CryptomatorFileProvider/FileProviderAdapter.swift

```diff
@@ -154,10 +163,15 @@ public class FileProviderAdapter: FileProviderAdapterType {
 					}
 					return completionHandler(nil, NSFileProviderError(.noSuchItem))
 				}
-				completionHandler(localItemImportResult.item, nil)
-
+				let localImportHandler: (Error?) -> Void = { error in
+					if let error = error {
+						completionHandler(nil, error)
+					} else {
+						completionHandler(localItemImportResult.item, nil)
+					}
+				}
 				// Network Stuff
-				self.uploadFile(taskRecord: localItemImportResult.uploadTaskRecord).then { item in
+				self.uploadFile(taskRecord: localItemImportResult.uploadTaskRecord, completionHandler: localImportHandler).then { item in
```

Comment:
Because we need to "inject" it before the uploadFile is done, i.e. before the upload file workflow gets scheduled but after the workflow gets created locally (as this is the completionHandler which is used to tell the Files app that the local handling for the upload has been completed).

However, we could split the called function into create upload task & workflow and scheduling the task.  And then simply call the completionHandler in the promise chain after creating the upload task & workflow.